### PR TITLE
Add Queue Component for 5.5

### LINKED ISSUE #83: Component updates for Laravel 5.5 : Queue
Complete update of the [Queue](https://github.com/mattstauffer/Torch/tree/master/4.2/queue) Component to Laravel 5.5 

See #73 for full details of this task/requirements.

### PR DESCRIPTION
**Pull request by mloberg**
Based off your queue branch, but with a few changes. Open to feedback on improving the submission.

Changes:

* I removed the use of configs as I felt that hid the setup of the Queue component rather than showing how to interact with it.
* I had issues getting Beanstalkd to work properly, so I went with Redis.

The workers were a bit challenging. It requires an ExceptionHandler, so I created one as an anonymous class. If there's an easier/better way to create one, let me know.

When running as a daemon, it is expecting an Application as it makes a call to `isDownForMaintenance`. To solve this I created an "App" and added the method. The daemon also ran fine, but then locked up.

Thank you for Torch! It's been a nice reference. Glad I can contribute in some small way.

Fixed #83

### CODE REVIEW

**Comment by mattstauffer** [comment]
Sorry, I have no idea why Nitpick is running on this codebase.

File: components/queue/index.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+$app->get('/redis/add', function () use ($container) {
+    $queue = $container['queue'];
+
+    $queue->connection('redis')->push('doThing', ['string' => 'redis-' . date('r')]);
+
+    echo 'Pushed an instance of doThing to redis.';
+});
+
+$app->get('/redis/work/worker', function() use ($container) {
```
[the remark this answers is not in the document]
Comment (mattstauffer) [reply components/queue/index.php]: OK yah this is a good code style note

File: components/queue/index.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    $events = $container['events'];
+    $handler = $container['exception.handler'];
+
+    $worker = new Worker($queue, $events, $handler);
+    $options = new WorkerOptions();
+
+    $worker->daemon('redis', 'default', $options);
+});
+
+$app->get('/redis/work/single', function() use ($container) {
```
[the remark this answers is not in the document]
Comment (mattstauffer) [reply components/queue/index.php]: Good code style note

File: components/queue/index.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    $handler = $container['exception.handler'];
+
+    $worker = new Worker($queue, $events, $handler);
+    $options = new WorkerOptions();
+
+    $worker->runNextJob('redis', 'default', $options);
+    echo 'Ran job';
+});
+
+class doThing
```
[the remark this answers is not in the document]
Comment (mattstauffer) [reply components/queue/index.php]: Ignore the first two note, but yah, this should be `DoThing`

**Comment by mloberg** [comment]
I'll try to find some time in the next couple days to address the comments and get beanstalkd working

**Comment by mattstauffer** [comment]
@mloberg I don't mind if we just make these few minor changes and merge it; then we can fix and merge beanstalkd at another time with less time pressure. Your call!

**Comment by mloberg** [comment]
@mattstauffer Fixed your style notes and updated the readme. Created #85 to finish beanstalkd examples.

Do you want me to squash these commits before you merge it?

**Comment by mattstauffer** [comment]
@mloberg awesome, reviewing now--thanks! No worries about squashing. I like seeing the real history.

**Comment by mattstauffer** [comment]
Fantastic--thank you!!

**Pull request merged**